### PR TITLE
Feature/document handles

### PR DESCRIPTION
**Pull request by kobizz**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/js/preview/preview.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		elementorCommon.api.run( 'editor/documents/switch', {
+			id: document.getSettings( 'id' ),
+			mode: 'autosave',
+		} );
+	}
+
+	onFrontendComponentsInit() {
+		this.createDocumentsHandles();
+
+		elementor.on( 'document:loaded', () => this.createDocumentsHandles() );
```
Comment (matipojo) [inline assets/dev/js/preview/preview.js]:
```suggestion
		elementor.once( 'document:loaded', () => this.createDocumentsHandles() );
```

File: assets/dev/js/preview/preview.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+export default class Preview extends elementorModules.ViewModule {
+	constructor() {
+		super();
+
+		elementorFrontend.on( 'components:init', () => this.onFrontendComponentsInit() );
+	}
+
+	createDocumentsHandles() {
+		jQuery.each( elementorFrontend.documentsManager.documents, ( index, document ) => {
+			const $documentElement = document.$element;
```
Comment (matipojo) [inline assets/dev/js/preview/preview.js]: Kit document has an $element?
It should be excluded.

File: modules/safe-mode/module.php [hunk 3]
```diff
@@ -20,7 +20,7 @@ class Module extends \Elementor\Core\Base\Module {
 	const DOCS_MU_PLUGINS_URL = 'https://go.elementor.com/safe-mode-mu-plugins/';
 	const DOCS_TRY_SAFE_MODE_URL = 'https://go.elementor.com/safe-mode/';
 
-	const EDITOR_NOTICE_TIMEOUT = 10000; /* ms */
+	const EDITOR_NOTICE_TIMEOUT = 30000; /* ms */
```
Comment (matipojo) [inline modules/safe-mode/module.php]: It's related to current PR?

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
See comments

File: assets/dev/js/preview/preview.js (hunk 1 above)
Comment (kobizz) [reply assets/dev/js/preview/preview.js]: It has to run on every document load

Comment (kobizz) [reply assets/dev/js/preview/preview.js]: We should check if handles already exist

**Review by matipojo — APPROVED** [review]

**Pull request merged**